src/main/java/seedu/address/logic/commands/MarkPaidCommand.java

```diff
@@ -49,13 +50,23 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Person personToMarkPayment = lastShownList.get(targetIndex.getZeroBased());
+
+        List<Participation> participationList = personToMarkPayment.getParticipation();
+        // UnEnroll student from current participation first
+        for (Participation p: participationList) {
+            new UnEnrollCommand(targetIndex, p.getTutorialSubject()).execute(model);
+        }
         Payment updatedPayment = calculatePayment(personToMarkPayment.getPayment(), fees);
         Person markedPerson = new Person(personToMarkPayment.getName(), personToMarkPayment.getPhone(),
                 personToMarkPayment.getEmail(), personToMarkPayment.getAddress(),
                 updatedPayment, personToMarkPayment.getParticipation(), personToMarkPayment.getTags());
 
         model.setPerson(personToMarkPayment, markedPerson);
         model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_PERSONS);
+        // Enroll student back
+        for (Participation p: participationList) {
+            new EnrollCommand(targetIndex, p.getTutorialSubject()).execute(model);
```

Comment:
This part too, it is the same issue as the one above, we should shift the updateFilteredPersonList below the for loop, and the participationList should not be looping over the original one, so maybe we can eithr create a duplicate within the method itself, or at getParticipation(), we can enforce it to return an immutable copy of it

Reply:
edited, works now